src/main/java/pl/psnc/dei/ui/components/batches/ComplexBatchImportComponent.java

```diff
@@ -0,0 +1,170 @@
+        selects.add(this.datasetSelect);
+        container.add(selects);
+
+        HorizontalLayout importName = new HorizontalLayout();
+        importName.addClassName("vertically-centered-row");
+        importName.add(this.nameTextFiled);
+        container.add(importName);
+
+        add(container);
+
+        HorizontalLayout uploads = new HorizontalLayout();
+        uploads.add(this.file);
+        add(uploads);
+
+        add(this.importButton);
+    }
+
+    private void initSelects() {
+        this.prepareProjectSelect();
+        this.prepareDatasetSelect();
+        this.prepareAggregatorSelect();
+    }
+
+    private void prepareAggregatorSelect() {
+        this.aggregatorSelect = new Select<>();
+        this.aggregatorSelect.setLabel("Select Aggregator");
+        this.aggregatorSelect.setItems(Arrays.stream(Aggregator.values()).filter(a -> a != Aggregator.UNKNOWN));
+        this.aggregatorSelect.setValue(Aggregator.getById(0));
+        this.aggregatorSelect.setEmptySelectionAllowed(false);
+    }
+
+    private void prepareProjectSelect() {
+        this.projectSelect = CommonComponentsFactory.getProjectSelector(this.projectsRepository);
+        this.projectSelect.setEmptySelectionAllowed(false);
+        this.projectSelect.addValueChangeListener(event -> {
+            Project project = this.projectSelect.getValue();
+            this.datasetSelect.setItems(this.datasetsRepository.findAllByProject(project));
+        });
+    }
+
+    private void prepareDatasetSelect() {
+        Project project = this.projectSelect.getValue();
+        this.datasetSelect =
+                CommonComponentsFactory.getDatasetSelector(
+                        (List<Dataset>) this.datasetsRepository.findAllByProject(project)
+                );
+    }
+
+    private void initUploads() {
+        this.prepareNameTextFiled();
+        this.prepareFileSelectButton();
+        this.prepareImportFile();
+        this.prepareImportButton();
+    }
+
+    private void prepareFileSelectButton() {
+        this.fileSelectButton = new Button();
+        this.fileSelectButton.setText("Upload");
+        this.fileSelectButton.setIcon(new Icon(VaadinIcon.CLOUD_UPLOAD_O));
+    }
+
+    private void prepareImportFile() {
+        this.memoryBuffer = new MemoryBuffer();
+        this.file = new Upload(this.memoryBuffer);
+        this.file.setMaxFiles(1);
+        this.file.setAcceptedFileTypes(".csv");
+        this.file.setDropAllowed(true);
+        this.file.setUploadButton(this.fileSelectButton);
+        this.file.addFileRejectedListener(
+                event -> Notification.show("File rejected. Reason: " + event.getErrorMessage(), 3000, Notification.Position.MIDDLE)
+        );
+    }
+
+    private void prepareNameTextFiled() {
+        this.nameTextFiled = new TextField();
+        this.nameTextFiled.setLabel("Import name");
+        this.nameTextFiled.addClassName("flex-1");
+        this.nameTextFiled.setMinLength(1);
```

Comment:
Import name should not be required,  we should accept values shorter than 1 char.

Reply:
allow empty input